File: java/Mahasiswa.java
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//membuat class Mahasiswa dengan menunjuk parentnya yaitu SivitasAkademik
class Mahasiswa extends SivitasAkademik{
    private String nim;
    private String nama;
    private String jenis_kelamin;
    private String fakultas;
    private String prodi;

    //membuat konstruktor sekaligus mengeset atribut private
    public Mahasiswa(String nim, String nama, String jenis_kelamin, String fakultas, String prodi){
        this.nim = nim;
        this.nama = nama;
        this.jenis_kelamin = jenis_kelamin;
        this.fakultas = fakultas;
        this.prodi = prodi;
    }

    //membuat getter dan setter
    //membuat konstruktor
    public Mahasiswa(){
    }
    //membuat fungsi untuk mengeset atribut private
    public void setNim(String nim){
        this.nim = nim;
    }
    public void setNama_Mahasiswa(String nama){
        this.nama = nama;
    }
    public void setJenis_Kelamin_Mahasiswa(String jenis_kelamin){
        this.jenis_kelamin = jenis_kelamin;
    }
    public void setFakultas(String fakultas){
        this.fakultas = fakultas;
    }
    public void setProdi(String prodi){
        this.prodi = prodi;
    }
    
    //membuat fungsi untuk me-return atribut private
    public String getNim(){
        return nim;
    }
    public String getNama_Mahasiswa(){
        return nama;
    }
    public String getJenis_Kelamin_Mahasiswa(){
        return jenis_kelamin;
    }
    public String getFakultas(){
        return fakultas;
    }
    public String getProdi(){
        return prodi;
    }
}

File: python/Mahasiswa.py
#Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
#Aamiin.

#Memeanggil class SivitasAakademik agar bisa dipakai
from SivitasAkademik import SivitasAkademik

#Membuat class Mahasiswa dengan SivitasAkademik sebagai parentnya
class Mahasiswa(SivitasAkademik):

    #variable/atribut private
    __nim = ""
    __nama = ""
    __jenis_kelamin = ""
    __fakultas = ""
    __prodi = ""

    #membuat konsruktor untuk mengubah/mengakses variabel/atribut private
    def __init__(self, nim, nama, jenis_kelamin, fakultas, prodi):
        self.__nim = nim
        self.__nama = nama
        self.__jenis_kelamin = jenis_kelamin
        self.__fakultas = fakultas
        self.__prodi = prodi
    
    #membuat fungsi untuk mengeset variabel/atribut private
    def set_nim(self, nim):
        self.__nim = nim
    def set_nama(self, nama):
        self.__nama = nama
    def set_jenis_kelamin(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
    def set_prodi(self, prodi):
        self.__prodi = prodi

    #membuat fungsi untuk me-return hasil proses konstruktor sebelumnya
    def get_nim(self):
        return self.__nim
    def get_nama(self):
        return self.__nama
    def get_jenis_kelamin(self):
        return self.__jenis_kelamin
    def get_fakultas(self):
        return self.__fakultas
    def get_prodi(self):
        return self.__prodi
    #pass agar bisa mengakses class parent
    pass
